### ISSUE TITLE
Stopped and continued recordings do not play

### ISSUE THREAD

**Issue by billie80** [yakovlevegor/DroidRec#36]
If a recording is stopped and continued the resulting videos do not play.
Videos recorded without pause play without problems.

**Comment by yakovlevegor** [comment]
Which video player are you using? What's your Android version?

**Comment by billie80** [comment]
Tried with VLC, the Player of Simple Gallery and the Player of AZ Screenrecorder. None of them is able to play the recording.
I am on Android 10.

**Comment by yakovlevegor** [comment]
Try specifying the recording folder again if the app crashes.

**Comment by billie80** [comment]
I am using 3.0 already and with this version there are no crashes. Just the issue with the non playable files.

**Comment by yakovlevegor** [comment]
Strange, because I don't experience any issues on an emulator running Android 10.

**Comment by yakovlevegor** [comment]
Could you post your codec list? You can find it in settings in choosing a codec menu.

**Comment by billie80** [comment]
![DroidRec](https://user-images.githubusercontent.com/9938306/206852970-428e7027-ed57-4d55-be0b-605149ace650.png)
This one?
Tested all the three of them. None produced playable results. With the second and third the application crashed after stopping the recording but it generated the files.

**Comment by yakovlevegor** [comment]
I've also tested on Google's codec, nothing went wrong. Seems like a very odd issue...

**Comment by yakovlevegor** [comment]
With #32 a temporary workaround for me was to set FPS to 30, though I doubt it's a liable solution.

**Comment by billie80** [comment]
This does not work for me. And isn't 30 the default frame rate?

**Comment by yakovlevegor** [comment]
No, the default tried framerate is what your screen can support. All codecs are tried to determine which one supports it. If none, it drops to 30.

**Comment by yakovlevegor** [comment]
I have no idea on this.

**Comment by yakovlevegor** [comment]
No, actually, I've got one. Mind testing it?
[app-test-dontstopdaemon.zip](https://github.com/yakovlevegor/DroidRec/files/10200071/app-test-dontstopdaemon.zip)

**Comment by billie80** [comment]
I tried it but unfortunately it did not solve the issue.

**Comment by billie80** [comment]
I can upload a sample file if it helps.
I just noticed that I can work around this issue if I pause the recording before stopping it. The the recording is playable. If I stop it directly it wont play.

**Comment by yakovlevegor** [comment]
>if I pause the recording before stopping it

Thanks, that's useful.

**Comment by yakovlevegor** [comment]
Maybe I could implement a workaround to pause the recording before stopping it.

**Comment by yakovlevegor** [comment]
Does this one solve it?
[app-test-autopause.zip](https://github.com/yakovlevegor/DroidRec/files/10200191/app-test-autopause.zip)

**Comment by billie80** [comment]
I am sorry but it doesn't. :-(
Even the workaround does not work anymore. It also crashes sometimes when pausing and stopping the recording.

**Comment by yakovlevegor** [comment]
I've added a new variable to pause only the writing process of the recording.
[app-test-pausewriting.zip](https://github.com/yakovlevegor/DroidRec/files/10200246/app-test-pausewriting.zip)

**Comment by billie80** [comment]
Unfortunately it is still the same. This seems really strange.

**Comment by yakovlevegor** [comment]
After pause the recorder uses a timestamp to continue recording to file. I've set it to update only by audio and video tracks, because it may interfere with the end of the record signal.
[app-test-modonrecord.zip](https://github.com/yakovlevegor/DroidRec/files/10200391/app-test-modonrecord.zip)

**Comment by billie80** [comment]
This time it looks better. All recordings play now.
Sometimes for the part before the pause only audio was playing but guess this was because the two segments only had 3 seconds each seconds. For longer segments this does not seem to occur.

**Comment by yakovlevegor** [comment]
I've been experiencing kind of a similar issue too, videos below 3 seconds are 0 bytes.

**Comment by yakovlevegor** [comment]
Though it shouldn't happen after pause. I've removed the writing process pause variable, just to test if it gets any better.
[app-test-fixedwithnopause.zip](https://github.com/yakovlevegor/DroidRec/files/10200469/app-test-fixedwithnopause.zip)

**Comment by yakovlevegor** [comment]
So everything is okay now with pause?

**Comment by yakovlevegor** [comment]
>Might be related to the codec used

Maybe, but I've experienced it when rotating the screen.

**Comment by billie80** [comment]
Yes just tested your latest version. This one also produces good results.

**Comment by yakovlevegor** [comment]
Great, so I'm preparing a new release. Thanks for all the testing trouble!

**Comment by billie80** [comment]
You are welcome.
Thank you for taking care of it so quickly.

**Comment by yakovlevegor** [comment]
Version [3.1](https://github.com/yakovlevegor/DroidRec/releases/tag/v3.1) is out. The fix is there.

**Comment by yakovlevegor** [comment]
The app crashes after pause with the fix, reopening.
